### ISSUE TITLE
Elastic Beanstalk: S3 client configured for "us-west-2" but the bucket "xyz" is in "us-west-2"

### ISSUE THREAD

**Issue by alexjurkiewicz** [travis-ci/dpl#1173]
We are using dpl v2 like this:

```yaml
      deploy:
        # Build every dev branch commit and upload to "Kounta Test" EB application
        - provider: elasticbeanstalk
          edge: true # opt in to dpl v2
          access_key_id:
            secure: *AWS_ACCESS_KEY_ID
          secret_access_key:
            secure: *AWS_SECRET_ACCESS_KEY
          region: us-west-2
          app: XXX
          env: Whatever # only_create_app_version is true
          only_create_app_version: true
          bucket: XXX
          bucket_path: XXX
          description: $TRAVIS_COMMIT_MESSAGE
          on:
            repo: XXX/XXX
            all_branches: true
```

We get the following error on deployment:

```
Install deployment dependencies

Installing gem dependencies: aws-sdk-elasticbeanstalk (~> 1.0), aws-sdk-s3 (~> 1.0), rubyzip (~> 1.2.2), pathspec (~> 0.2.1)

Authenticate deployment
Using Access Key: AWS*****************

Setup deployment
Setting the build environment up for the deployment
Using Access Key: AWS*****************

Prepare deployment

Run deployment
S3 client configured for "us-west-2" but the bucket "XXX" is in "us-west-2"; Please configure the proper region to avoid multiple unnecessary redirects and signing attempts
(Aws::S3::Errors::BadRequest)
failed to deploy

6
```

(You can see the private build error here: https://travis-ci.com/kounta/kounta/jobs/291196364)

This problem is also occurring in dpl 1. I tried upgrading to v2 to see if anything changed and to report this bug.

**Comment by alexjurkiewicz** [comment]
I set `debug: true` and re-ran without any extra output.

**Comment by alexjurkiewicz** [comment]
Travis support pointed at travis-ci/travis-ci#9747, which is the same issue and suggests my IAM key is wrong.

**Comment by alexjurkiewicz** [comment]
I figured out what the issue was. I was using the same encrypted blob I used for our global environment variables. eg `echo AWS_ACCESS_KEY_ID=xxx | travis encrypt`. But these parameters only require the string, eg `echo xxx | travis encrypt`.

So the effective config was like:

```yaml
access_key_id:
  AWS_ACCESS_KEY_ID: xxx
```

This seems to come up a few times a year, so it would be nice for some sanity detection of these values to detect this common error case 🙏
